data_processing.ipynb

```diff
@@ -748,6 +748,236 @@
     "print(test_gpac_data['data']['label'].value_counts())"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pywt\n",
+    "import scipy\n",
+    "\n",
+    "def do_cwt(X,freq,fs,wavename = 'cmor5-0.8125'):\n",
+    "    \"\"\"Computes complex wavelet transform for EEG data\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    X : input data\n",
+    "    freq : input frequency range eg. 0.25:0.25:50\n",
+    "    fs : sampling frequency\n",
+    "    wavename : The default is 'cmor5-0.8125'.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    cwt complex coefficients\n",
+    "    frequency range\n",
+    "    time\n",
+    "    \"\"\"\n",
+    "    T = 1.0/fs\n",
+    "    t = np.arange(0,len(X)*T,T)\n",
+    "    a_coef = pywt.centrfrq(wavename)* (1.0/(freq*T))\n",
+    "    \n",
+    "    # calculate the wavelet coefficients (distance x frequency x time)\n",
+    "    cwt_c = pywt.cwt(X,a_coef,wavename, method='fft')\n",
+    "    return cwt_c[0], freq, t\n",
+    "\n",
+    "def get_all_cwts(X, lf, hf, fs, wavename='cmor5-0.8125'):\n",
```

Comment:
wavename here is a different string than in the do_cwt calls. Is this intended? Is this just copied over from what Berj/Daniel gave us?